DATA/production/common/setVarsFromALIEN_PROC_ID.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+# Script to set a couple of variables that depend on ALIEN_PROC_ID.
+# Since several scripts might need this, it is in a separate script.
+
+# let's get the last 10 (for INT) and 5 (for int16) digits of ALIEN_PROC_ID, to be passed to define NUMAID and shm-segment-id via O2JOBID, which are int and int16 respectively. Then we make them an int or int16
+ALIEN_PROC_ID_MAX_NDIGITS_INT32=10
+ALIEN_PROC_ID_MAX_NDIGITS_INT16=9
+ALIEN_PROC_ID_LENTGH=`echo $ALIEN_PROC_ID | wc -l`
+ALIEN_PROC_ID_OFFSET_INT32=$((ALIEN_PROC_ID_LENTGH - ALIEN_PROC_ID_MAX_NDIGITS_INT32))
+ALIEN_PROC_ID_OFFSET_INT16=$((ALIEN_PROC_ID_LENTGH - ALIEN_PROC_ID_MAX_NDIGITS_INT16))
+# lets's take the last 10 (int) or 5 (int16) digits
+ALIEN_PROC_ID_OFFSET_INT32=${ALIEN_PROC_ID:$((ALIEN_PROC_ID_OFFSET_INT32))}
+ALIEN_PROC_ID_OFFSET_INT16=${ALIEN_PROC_ID:$((ALIEN_PROC_ID_OFFSET_INT16))}
+# let's make them int32 or int16
+export O2JOBID=$((ALIEN_PROC_ID_OFFSET_INT32 & 0x7fffffff - 1)) # subtracting 1 since later we add the NUMAID which can be up to 1
+export SHMEMID=$((ALIEN_PROC_ID_OFFSET_INT16 & 0x7fff - 1)) # subtracting 1 since later we add the NUMAID which can be up to 1
```

Comment:
@chiarazampolli : I think this is wrong, you cannot simply subtract one. If the orignal id was 0, you'll end up with -1 :)
I think I'd use `ALIEN_PROC_ID_OFFSET_INT16 % 0x7ff0`, i.e. modulo with a slightly smaller module. This will guarantee we can add 0 -- 15 on top afterwards.

also, SHMEID is a bit too generic, and not clear where it comes from.
Could we call it O2JOBSHMID?

Reply:
You are right!!! Implemented.